flake.nix

```diff
@@ -0,0 +1,68 @@
+{
+  description = "Rust development dev shell";
+
+  inputs = {
+    nixpkgs.url      = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-utils.url  = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, rust-overlay, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+      in
+      with pkgs;
+      {
+        devShells.default = mkShell {
+          buildInputs = [
+            openssl
+            pkg-config
+            eza
+            fd
+            lldb
+           # rustup
+            clang
+            cmake   # Example: build system
+            (pkgs.rust-bin.nightly.latest.rust.override {
+              extensions = ["rust-src" "clippy" "rustfmt" "rust-analysis" "rustc" "miri" "rust" "rust-std" "cargo" "rust-analyzer"];
+            })
+            cargo-audit
+          ];
+
+          shellHook = ''
+            echo "Dev shell is starting..."
+            export CC=$(which gcc)  # Set the C compiler
+            export CXX=$(which g++) # Set the C++ compiler
+            alias ls=eza
+            alias find=fd
+            export PATH="$HOME/.cargo/bin:$PATH"
+            cargo install cargo-xtask
```

Comment:
[It's](https://github.com/tiiuae/ebpf-firewall/pull/3/files#r1709295801) from here. Let's have consistent dependency install via flake+rust instead.